src-built-in/components/appMarket/src/components/SearchBar.jsx

```diff
@@ -0,0 +1,100 @@
+/*!
+* Copyright 2017 by ChartIQ, Inc.
+* All rights reserved.
+*/
+/**
+ * This component is the name of a component and a pin that will pin that component to all toolbars.
+ *
+ */
+import React, { Component } from "react";
+
+//components
+import Toast from './Toast';
+// import Tag from './Tag';
+import Tag from '../../../shared/Tag';
+import TagsMenu from '../../../shared/TagsMenu';
+
+class SearchBar extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			searchValue: "",
+			tagSelectorOpen: false
+		};
+		this.bindCorrectContext();
+	}
+	bindCorrectContext() {
+		this.changeSearch = this.changeSearch.bind(this);
+		this.toggleTagSelector = this.toggleTagSelector.bind(this);
+		this.selectTag = this.selectTag.bind(this);
+		this.removeTag = this.removeTag.bind(this);
+	}
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.activeTags.length === 0 && !nextProps.backButton) {
+			this.setState({
+				searchValue: ""
+			});
+		}
+	}
+	changeSearch(e) {
+		this.setState({
```

Comment:
The store doesn't/shouldn't need to keep track of an inputs text. That's not what stores are for. The state in this case is simply the text inside the input when the component is mounted. I can move it, but that seems like bad component management to me. Just because it _CAN_ go in a store, doesn't mean it should.

Reply:
I hear you. I usually put that stuff in a store because I inevitably need to modify the text field from some other component (e.g., a filter, share client, etc).